### ISSUE TITLE
Off-axis parabolic mirror

### ISSUE THREAD

**Issue by ppugpug** [opticspy/lightpipes#39]
I am using LightPipes for THz optics design. Is there a way to use off-axis parabolic mirrors in LightPipes?

Best regards,
Prach

**Comment by ldoyle** [comment]
Hi Prach,

that is a good question. Right now, there is no function to do this and I know of no example code which achieved this with LightPipes.
However, to first approximation, an OAP is "just a lens" which changes the direction of the optical axis. In a recent project, I used LightPipes to compare the focus generated by an OAP with the wavefront measured in the recollimated beam after the OAP, and it worked very well. The OAP in this case is 90deg off-axis angle, f/2 focusing (28cm beam, 50cm eff. focal length).
I am aware that there are slight asymmetries in the field components in the focus of an OAP, but since LightPipes only uses a scalar field these might not be calculable correctly anyway.

I would be interested if you have any literature comparing the maths for a simple lens and an OAP. I came across these two sources in the past:
* https://link.springer.com/article/10.1007/s00340-018-7084-9 They show how to numerically calculate all vectorial E and B components of the field, but don't detail it out a 100% as the paper is more about the usage. Maybe try the references therein.
* https://www.lighttrans.com/use-cases/application-use-cases/high-na-focusing-by-off-axis-parabolic-mirror.html This program I  know has an OAP component and full vectorial field, the example images in the PDF show the magnitude of the fields in Ex, Ey, Ez.

Hope this helps.

Regards, Leonard
